src/components/search/search-result.component.tsx

```diff
@@ -67,26 +68,25 @@ const SearchResultComponent: React.FC<SearchResultProps> = ({ isAnonymous, user,
 
   useEffect(() => {
     // list all transaction user id as param
-    checkFriendStatus(users.map(item => item.id));
+    if (user) checkFriendStatus([user.id]);
   }, []);
 
   const router = useRouter();
   const redirectToProfilePage = (url: string) => {
     router.push(`/${url}`);
   };
 
-  const getFriendStatus = (user: User): FriendStatus | null => {
+  const getFriendStatus = (friendUser: User): FriendStatus | null => {
     const found = friended.find(friend => {
```

Comment:
```suggestion
    const found = friendsList.find(friend => {
```

Reply:
i would rename `friend` to `friendOrFriendRequested`

Reply:
its ok

Reply:
> 

mungkin untuk ganti friended to friendList harus merubah state di contextnya sekalian later,

Reply:
setuju sama @rei-gun , `found` diganti dulu baru kita resolve @immhrmwn